compiler/rustc_lint/src/levels.rs

```diff
@@ -344,6 +352,20 @@ impl<'s> LintLevelsBuilder<'s> {
                     self.store.check_lint_name(&name, tool_name, self.registered_tools);
                 match &lint_result {
                     CheckLintNameResult::Ok(ids) => {
+                        // This checks for instances where the user writes `#[expect(unfulfilled_lint_expectations)]`
+                        // in that case we want to avoid overriding the lint level but instead add an expectation that
+                        // can't be fulfilled. The lint message will include an explanation, that the
+                        // `unfulfilled_lint_expectations` lint can't be expected.
+                        if let Level::Expect(expect_id) = level {
+                            let is_unfulfilled_lint_expectations = match ids {
```

Comment:
Please add a comment that matching here against `[lint]` only works, because this lint is currently not in a lint group.

Reply:
Good hint, thank you for the review!